### PR TITLE
fix(share/discovery): fixes from Advertise logic audit

### PR DESCRIPTION
**Pull request by Wondertan**
* Use only TTL provided by us.
* Don't wait 8hs upon error
* Increase the advertise timeout

### CODE REVIEW

File: share/availability/discovery/discovery.go [hunk 1]
```diff
@@ -151,32 +147,38 @@ func (d *Discovery) Peers(ctx context.Context) ([]peer.ID, error) {
...
 	defer timer.Stop()
 	for {
-		ttl, err := d.disc.Advertise(ctx, rendezvousPoint)
+		_, err := d.disc.Advertise(ctx, rendezvousPoint)
 		if err != nil {
-			log.Debugf("Error advertising %s: %s", rendezvousPoint, err.Error())
 			if ctx.Err() != nil {
 				return
 			}
+			log.Debugf("error advertising %s: %s", rendezvousPoint, err.Error())
```
Comment (walldiss) [inline share/availability/discovery/discovery.go:162]: Maybe Warn?

File: share/availability/discovery/discovery.go [hunk 2]
```diff
@@ -151,32 +147,38 @@ func (d *Discovery) Peers(ctx context.Context) ([]peer.ID, error) {
...
 
+			errTimer := time.NewTimer(time.Minute)
 			select {
-			case <-timer.C:
+			case <-errTimer.C:
+				errTimer.Stop()
+				if !timer.Stop() {
+					<-timer.C
+				}
 				timer.Reset(d.params.AdvertiseInterval)
```
Comment (walldiss) [inline share/availability/discovery/discovery.go:171]: Does it make sense to reset advertiseTimer only after successful advertise?

Comment (Wondertan) [reply share/availability/discovery/discovery.go:171]: Yes, makes sense

File: share/availability/discovery/discovery.go (hunk 1 above)
Comment (Wondertan) [reply share/availability/discovery/discovery.go]: 👌🏻

**Review by renaynay — APPROVED** [review]

**Pull request merged**